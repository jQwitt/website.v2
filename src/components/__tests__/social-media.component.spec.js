import React from 'react';
import { render, cleanup } from '@testing-library/react';
import SocialMedia from '../social-media.component';

afterEach(cleanup);
describe('SocialMedia renders correctly', () => {
  test('snapshot', () => {
    const { asFragment } = render(<SocialMedia />);
    expect(asFragment()).toMatchSnapshot();
  });
  it('conditionally justifies content', () => {
    const data = {
      justifyContent: ['start', 'center', 'end'],
    };
    data.justifyContent.forEach((cl) => {
      const { container } = render(<SocialMedia justifyContent={cl} />);
      expect(
        container.firstChild.classList.contains(`justify-content-${cl}`),
      ).toBe(true);
    });
  });

  it('conditionally sizes icons', () => {
    const data = {
      size: ['xsmall', 'small', 'medium', 'large', 'xlarge', 'jumbo'],
    };
    data.size.forEach((cl) => {
      const { container } = render(<SocialMedia size={cl} />);
      expect(container.querySelector('i').classList.contains(cl)).toBe(true);
    });
  });
});
