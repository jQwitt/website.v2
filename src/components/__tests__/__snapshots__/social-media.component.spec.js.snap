// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SocialMedia renders correctly snapshot 1`] = `
<DocumentFragment>
  <div
    class="social-media-links d-flex justify-content-center"
  >
    <a
      href="https://www.linkedin.com/in/joseph-witt-52a960171/"
    >
      <i
        class="flaticon-linkedin medium"
      />
    </a>
    <a
      href="mailto:jackwittbiz@gmail.com"
    >
      <i
        class="flaticon-email medium"
      />
    </a>
    <a
      href="https://github.com/jQwitt"
    >
      <i
        class="flaticon-browser medium"
      />
    </a>
  </div>
</DocumentFragment>
`;
